const {
  getAuth,
  createUserWithEmailAndPassword,
  signOut,
} = require("firebase/auth");
const auth = getAuth();

exports.signup = async (req, res) => {
  try {
    const user = await createUserWithEmailAndPassword(
      auth,
      req.body.email,
      req.body.password
    );
    res.status(200).json({
      success: true,
      data: user,
    });
  } catch (err) {
    res.status(400).json({
      success: false,
      error: err.message,
    });
  }
};
exports.signout = async (req, res) => {
  try {
    const user = await signOut(auth);
    res.status(200).json({
      success: true,
      data: user,
    });
  } catch (err) {
    res.status(400).json({
      success: false,
      error: err.message,
    });
  }
};
// createUserWithEmailAndPassword(auth, email, password)
//   .then((userCredential) => {
//     // Signed in
//     const user = userCredential.user;
//     // ...
//   })
//   .catch((error) => {
//     const errorCode = error.code;
//     const errorMessage = error.message;
//     // ..
//   });
// signOut(auth)
//   .then(() => {
//     // Sign-out successful.
//   })
//   .catch((error) => {
//     // An error happened.
//   });
